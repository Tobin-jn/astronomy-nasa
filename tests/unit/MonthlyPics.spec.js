import { shallowMount, mount } from "@vue/test-utils";
import MonthlyPics from "@/views/MonthlyPics.vue";

describe('MonthlyPics', () => {
  it('is a Vue instance', () => {
    const wrapper = mount(MonthlyPics)
    expect(wrapper.isVueInstance()).toBeTruthy()
  });
  it('renders correctly', () => {
    const wrapper = mount(MonthlyPics)
    expect(wrapper.element).toMatchSnapshot()
  });
  it('should return the date in the correct form', () => {
    const wrapper = mount(MonthlyPics)
    expect(wrapper.vm.getDay('2019-01-10')).toEqual('January 10, 2019')
  });
});