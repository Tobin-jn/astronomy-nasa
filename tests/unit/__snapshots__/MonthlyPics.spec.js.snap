// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthlyPics renders correctly 1`] = `
<div
  class="monthlyPics"
>
  <h1>
     Astronomy Pictures
  </h1>
   
  <router-link
    to="/"
  >
    <img
      alt="Home Icon"
      class="home-btn"
      src="../assets/home-icon.png"
    />
  </router-link>
   
  <div
    class="card-container"
  >
    <div
      class="day-card"
    >
      <h4>
         January 01, 2019 
      </h4>
       
      <img
        alt="Picture of the Day"
        class="day-image"
        src="https://apod.nasa.gov/apod/image/1812/WitchHeadNebula_Mishra_2725.jpg"
        width="200"
      />
    </div>
    <div
      class="day-card"
    >
      <h4>
         January 02, 2019 
      </h4>
       
      <img
        alt="Picture of the Day"
        class="day-image"
        src="https://apod.nasa.gov/apod/image/1901/sombrero_spitzer_3000.jpg"
        width="200"
      />
    </div>
    <div
      class="day-card"
    >
      <h4>
         January 03, 2019 
      </h4>
       
      <img
        alt="Picture of the Day"
        class="day-image"
        src="https://apod.nasa.gov/apod/image/1901/orionred_WISEantonucci_1824.jpg"
        width="200"
      />
    </div>
    <div
      class="day-card"
    >
      <h4>
         January 04, 2019 
      </h4>
       
      <img
        alt="Picture of the Day"
        class="day-image"
        src="https://apod.nasa.gov/apod/image/1901/20190102UltimaThule-pr.png"
        width="200"
      />
    </div>
    <div
      class="day-card"
    >
      <h4>
         January 05, 2019 
      </h4>
       
      <img
        alt="Picture of the Day"
        class="day-image"
        src="https://apod.nasa.gov/apod/image/1901/UT-blink_3d_a.gif"
        width="200"
      />
    </div>
    <div
      class="day-card"
    >
      <h4>
         January 06, 2019 
      </h4>
       
      <img
        alt="Picture of the Day"
        class="day-image"
        src="https://apod.nasa.gov/apod/image/1901/Jan3yutu2CNSA.jpg"
        width="200"
      />
    </div>
    <div
      class="day-card"
    >
      <h4>
         January 07, 2019 
      </h4>
       
      <img
        alt="Picture of the Day"
        class="day-image"
        src="https://apod.nasa.gov/apod/image/1901/vltlaser_beletsky_1400.jpg"
        width="200"
      />
    </div>
    <div
      class="day-card"
    >
      <h4>
         January 08, 2019 
      </h4>
       
      <img
        alt="Picture of the Day"
        class="day-image"
        src="https://apod.nasa.gov/apod/image/1901/Geminids46P_jcc_2000.jpg"
        width="200"
      />
    </div>
    <div
      class="day-card"
    >
      <h4>
         January 09, 2019 
      </h4>
       
      <img
        alt="Picture of the Day"
        class="day-image"
        src="https://apod.nasa.gov/apod/image/1812/CarinaNebulaGerminiani.jpg"
        width="200"
      />
    </div>
  </div>
   
  <button
    class="daily-pic-btn"
  >
    <router-link
      to="/DailyPic"
    >
      Picture of the Day
    </router-link>
  </button>
</div>
`;
